sdk/core/azure-core/src/http/url.cpp

```diff
@@ -177,6 +177,25 @@ void Url::AppendQueryParameters(const std::string& query)
   }
 }
 
+std::string Url::GetUrlWithoutQuery() const
+{
+  std::string full_url;
+  if (!m_scheme.empty())
+  {
+    full_url += m_scheme + "://";
+  }
+  full_url += m_host;
+  if (m_port != 0)
+  {
+    full_url += ":" + std::to_string(m_port);
+  }
+  if (!m_encodedPath.empty())
+  {
+    full_url += "/";
+  }
```

Comment:
Do we need to add `/` depending if the url has or not a path?
This integrates fine with the current implementation for calling GetUrlWithoutQuery() + GetRelativeUrl() for the absolute URL.
However, if you are calling `GetUrlWithoutQuery()` for logging, are expecting show the path as well?  
Right now, the path won't be displayed and only `/` when it has a path or nothing for no path.

Is this what you expect for logging?

For example, a request to `http://host.com/resource/foo` would be logged as `http://host.com/`
and
`http://host.com` is logged as `http://host.com`

Reply:
If that's the behavior, this is not what I expect. I'll fix.